packages/back-end/src/routers/metric-group/metric-group.controller.ts

```diff
@@ -0,0 +1,187 @@
+import { getDataSourceById } from "back-end/src/models/DataSourceModel";
+import { removeMetricFromExperiments } from "back-end/src/models/ExperimentModel";
+
+export const getMetricGroups = async (
+  req: AuthRequest,
+  res: Response<{ status: 200; metricGroups: MetricGroupInterface[] }>
+) => {
+  const context = getContextFromReq(req);
+
+  const metricGroups = await context.models.metricGroups.getAll();
+  res.status(200).json({
+    status: 200,
+    metricGroups,
+  });
+};
+
+export const postMetricGroup = async (
+  req: AuthRequest<CreateMetricGroupProps>,
+  res: Response<{ status: 200; metricGroup: MetricGroupInterface }>
+) => {
+  const data = req.body;
+  const context = getContextFromReq(req);
+
+  if (!context.permissions.canCreateMetricGroup()) {
+    context.permissions.throwPermissionError();
+  }
+
+  const datasourceDoc = await getDataSourceById(context, data.datasource);
+  if (!datasourceDoc) {
+    throw new Error("Invalid data source");
+  }
+
+  const baseMetricGroup: Omit<
+    MetricGroupInterface,
+    "id" | "organization" | "dateCreated" | "dateUpdated"
+  > = {
+    ...data,
+    owner: data.owner || "",
+    description: data.description || "",
+    tags: data.tags || [],
+    projects: data.projects || [],
+    archived: data.archived || false,
+  };
+
+  const doc = await context.models.metricGroups.create(baseMetricGroup);
+
+  res.status(200).json({
+    status: 200,
+    metricGroup: doc,
+  });
+};
+
+export const putMetricGroup = async (
+  req: AuthRequest<CreateMetricGroupProps, { id: string }>,
+  res: Response<{ status: 200 }>
+) => {
+  const data = req.body;
+  const context = getContextFromReq(req);
+  const { org } = context;
+
+  const metricGroup = await context.models.metricGroups.getById(req.params.id);
+  if (!metricGroup) {
+    throw new Error("Could not find metric group with that id");
+  }
+  if (org.id !== metricGroup.organization) {
+    throw new Error("You don't have access to that metric group");
+  }
+
+  if (!context.permissions.canUpdateMetricGroup()) {
+    context.permissions.throwPermissionError();
+  }
+
+  const datasourceDoc = await getDataSourceById(
+    context,
+    data?.datasource || metricGroup.datasource
+  );
+  if (!datasourceDoc) {
+    throw new Error("Invalid data source");
+  }
+  await context.models.metricGroups.updateById(req.params.id, data);
+
+  res.status(200).json({
+    status: 200,
+  });
+};
+
+export const deleteMetricGroup = async (
+  req: AuthRequest<null, { id: string }>,
+  res: Response<{ status: 200 }>
+) => {
+  const context = getContextFromReq(req);
+
+  if (!context.permissions.canDeleteMetricGroup()) {
+    context.permissions.throwPermissionError();
+  }
+
+  const metricGroup = await context.models.metricGroups.getById(req.params.id);
+
+  if (!metricGroup) {
+    throw new Error("Could not find the metric group");
+  }
+
+  // should we delete all references to this metric group in the experiments?
```

Comment:
i think no. eventually we will have metric versioning, and could show better status messages